pkg/providers/snow/objects.go

```diff
@@ -57,18 +86,17 @@ func WorkersSpec(ctx context.Context, log logr.Logger, spec *cluster.Spec, kubeC
 	return w, nil
 }
 
-// WorkersObjects generates all the objects that compose a Snow specific CAPI spec for the worker nodes of an eks-a cluster.
-func WorkersObjects(ctx context.Context, log logr.Logger, clusterSpec *cluster.Spec, kubeClient kubernetes.Client) ([]kubernetes.Object, error) {
-	w, err := WorkersSpec(ctx, log, clusterSpec, kubeClient)
-	if err != nil {
-		return nil, err
+func workersIPPools(clusterSpec *cluster.Spec) CAPASIPPools {
+	capasPools := CAPASIPPools{}
+	for _, workerNodeGroupConfig := range clusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations {
```

Comment:
you are already iterating over `WorkerNodeGroupConfigurations` in `WorkersSpec` to build the worker groups, could we take advantage of that loop to build the IPPools?

Reply:
The IPPools are needed to build workerMachineTemplates and kubeadmConfigTemplates. I don't think it will work to use the loop in the workersSpec. I can certainly restruct the WorkersSpec and  WorkersMachineAndConfigTemplate to maybe only have one loop but I don't think it's necessary.

Reply:
Got you. Refactoring the existing code seems out of scope for this PR. I'll leave it up to you if you want to do that later or not.

I think eventually we should unify how all providers generate the capi spec. Maybe we should start thinking about how that is gonna look and then refactor this once we have that common vision defined.

Reply:
I think we talked abt it offline, I agree we should have consistent logic. I will have followup PR to refactor how snow worker spec gets generated to align with the clusterapi API. It's in my TODO